semgrep-core/src/parsing/pfff/scala_to_generic.ml

```diff
@@ -121,10 +121,14 @@ let v_dotted_name_of_stable_id (v1, v2) =
   let id = id_of_simple_ref v1 in
   id :: v2
 
-let rec v_import_expr tk (v1, v2) =
-  let module_name = G.DottedName (v_dotted_name_of_stable_id v1) in
-  let v2 = v_import_spec v2 in
-  v2 tk module_name
+let rec v_import_expr tk import_expr =
+  match import_expr with
+  | Left id ->
+      [ { G.d = G.ImportFrom (tk, DottedName [], id, None); d_attrs = [] } ]
```

Comment:
Hmm, there's an unenforced invariant that `dotted_ident`s have at least one element. Using `ImportAll` instead of `ImportFrom` wouldn't require you to create a malformed `dotted_ident`. I assume you tried this and it didn't match, though?

Take a look at `Normalize_generic.ml` and `m_directive` in `Generic_vs_generic.ml`. See if you can structure this in such a way that:
1. No malformed dotted_idents are created
2. `import $X` matches various imports that Scala allows (and add tests)

You might need to tweak some of the matching code to make this happen. I haven't looked into this very closely, but I'm happy to help talk things over if you would like.

Reply:
Actually, it looks like `ImportAll` does not get normalized, and there is a comment that it should not get normalized. I'm not sure what the best way forward here is. Maybe look to see what is done for Java, which is very similar?

Reply:
It seems to be consistent with what Java does:
```java
import $X;
```
translates to
`Pr([DirectiveStmt({d_attrs=[]; d=ImportFrom((), DottedName([]), ("$c", ()), None); })])`

So it looks like `DottedName`s can be empty anyways.